[docs] Add `hooks` (#42355)
expo/expo: docs/pages/eas/workflows/pre-packaged-jobs.mdx

| ios_bundle_identifier | string | The iOS bundle identifier of the submitted build. |
| android_package_id    | string | The Android package ID of the submitted build.    |

#### Hooks

> Hooks are supported for this job. Available hooks:

Reviewer: ```suggestion
Following hooks are supported for the Submit job:
```

"This" signifying the Submit job is bit far away since this is a separate subsection that doesn't immediately follows the main section. I think just calling out job name job is a small readability improvement.

| device_identifier            | string or `{ android?: string, ios?: string }` object | Optional. Device identifier to use for the tests. You can also use a single-value expression like `pixel_6`, `iPhone 16 Plus` or `${{ needs.build.outputs.platform == "android" ? "pixel_6" : "iPhone 16 Plus" }}` and an object like `device_identifier: { android: "pixel_6", ios: "iPhone 16 Plus" }`. Note that iOS devices availability differs across runner images. A list of available devices can be found in the jobs logs. |
| skip_build_check             | boolean                                               | Optional. Skip validation of the build (whether an iOS build is a simulator build). Defaults to false.                                                                                                                                                                                                                                                                                                                                |

#### Hooks

> Hooks are supported for this job. Available hooks:

Reviewer: ```suggestion
Following hooks are supported for the Maestro job:
```

</Collapsible>

<Collapsible summary="Generate Maestro flows before running tests">

> This workflow uses a hook to generate the `maestro_tests` directory right before Maestro starts running tests.

Reviewer: ```suggestion
The following workflow uses a hook to generate the `maestro_tests` directory right before Maestro starts running tests.
```

</Collapsible>

<Collapsible summary="Report Maestro artifacts to an external service">

> This workflow saves screenshots and recordings to `MAESTRO_TESTS_DIR`, then runs a script after the tests finish to upload those files or send a summary to an external service.

Reviewer: ```suggestion
The following workflow saves screenshots and recordings to `MAESTRO_TESTS_DIR`, then runs a script after the tests finish to upload those files or send a summary to an external service.
```

If we can provide an example of external service here that'll be great for clarity.

> **important** You need to either set `maestro_api_key` parameter or `MAESTRO_CLOUD_API_KEY` environment variable in the job environment. Go to "Settings" on [Maestro Cloud](https://app.maestro.dev/) to generate an API key and then to [Environment variables](https://expo.dev/accounts/[account]/projects/[project]/environment-variables) to add it to your project.

> Hooks are supported for this job. Available hooks:

Reviewer: ```suggestion
Following hooks are supported for the Maestro Cloud job:
```
